ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java

```diff
@@ -166,10 +166,10 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     final var block =
         new Block(newBlockHeader, new BlockBody(transactions, Collections.emptyList()));
 
-    if (mergeContext.isSyncing() || parentHeader.isEmpty()) {
+    if (mergeContext.map(c -> c.isSyncing()).orElse(Boolean.TRUE) || parentHeader.isEmpty()) {
```

Comment:
This is kinda hard to read. It looks like it evaluates to a no-op, does anyone else find this harder to read?

Reply:
I can see how it is harder to read, it is a side effect of making mergeContext an Optional.  looking for an approach that is easier on the 👀

Reply:
Supplier/function reference ftw 👍